### PR TITLE
feat: swappable http client

### PR DESCRIPTION
**Pull request by peterdeme**
Swappable http client

### CODE REVIEW

File: stream_chat/async_chat/client.py [hunk 1]
```diff
@@ -49,6 +49,14 @@ def __init__(
             connector=aiohttp.TCPConnector(keepalive_timeout=59.0),
         )
 
+    def set_http_session(self, session: aiohttp.ClientSession) -> None:
```
Comment (gumuz) [inline stream_chat/async_chat/client.py:52]: wouldn't it be nice to have some kind of interface instead of these 2 concrete session implementations?

Comment (gumuz) [reply stream_chat/async_chat/client.py:52]: or is this overkill for what we're trying to achieve here

Comment (peterdeme) [reply stream_chat/async_chat/client.py:52]: the problem is with typing. one of them is using `requests` the other is `aiohttp`

File: stream_chat/async_chat/client.py [hunk 2]
```diff
@@ -49,6 +49,14 @@ def __init__(
...
+        """
+        You can use your own `aiohttp.ClientSession` instance. This instance
+        will be used for underlying HTTP requests.
+        Make sure you set up a `base_url` for the session.
```
Comment (gumuz) [inline stream_chat/async_chat/client.py:56]: do we need a similar instruction for the sync/requests version of this method?

**Review by gumuz — APPROVED** [review]

File: stream_chat/async_chat/client.py (hunk 2 above)
Comment (peterdeme) [reply stream_chat/async_chat/client.py:56]: no, we use the full url there since it cannot be configured in `requests`

**Pull request merged**